File: Backup/src/passwordStrengthTester.js
/**
 * @param {string} password
 *
 * @description Avalia a força de uma senha e retorna o tipo de força da senha.
 *
 * @returns {string} O tipo de força da senha ('veryWeak', 'weak', 'regular', 'strong' ou 'veryStrong').
 *
 * @example
 * passwordStrengthTester('123'); // Output: 'veryWeak'
 *
 * @example
 * passwordStrengthTester('abc'); // Output: 'weak'
 *
 * @example
 * passwordStrengthTester('abc123'); // Output: 'regular'
 *
 * @example
 * passwordStrengthTester('Abc123!'); // Output: 'strong'
 *
 * @example
 * passwordStrengthTester('SuperSecurePassword123!'); // Output: 'veryStrong'
 *
 *
 * @returns {string} O tipo de força da senha ('veryWeak', 'weak', 'regular', 'strong' ou 'veryStrong').
 */
function passwordStrengthTester(password) {
  if (typeof password !== 'string') throw new TypeError('The input should be a string.');
  // Check de comprimento da senha
  const passwordLength = password.length;
  let strengthType;

  // Critérios para classificar a senha
  if (passwordLength < 6) {
    strengthType = 'veryWeak';
  } else if (passwordLength < 8) {
    strengthType = 'weak';
  } else if (passwordLength < 10) {
    strengthType = 'regular';
  } else if (
    /[A-Z]/.test(password)
    && /[!@#$%^&*(),.?":{}|<>]/.test(password)
    && /\d/.test(password)
    && /[a-zA-Z]/.test(password)
  ) {
    strengthType = 'veryStrong';
  } else {
    strengthType = 'strong';
  }

  return strengthType;
}
module.exports = passwordStrengthTester;

File: packages/python/src/multiform_validator/passwordStrengthTester.py
def passwordStrengthTester(password):
    if not isinstance(password, str):
        raise TypeError('The input should be a string.')

    password_length = len(password)
    strength_type = 'unknow'

    if password_length <= 5 and password.isdigit():
        strength_type = 'veryWeak'
    elif (password_length <= 5 and password.isalnum()) or \
         (6 <= password_length <= 7 and password.isalnum()) or \
         (password_length < 10 and len(set(password)) <= 2) or \
         (5 <= password_length <= 8 and password.isdigit()):
        strength_type = 'weak'
    elif len(set(password)) == 1 and password_length > 10:
        strength_type = 'regular'
    elif password_length > 16 or \
         (8 <= password_length and any(c.isupper() for c in password) and \
          any(c.islower() for c in password) and \
          any(c.isdigit() for c in password) and \
          any(c.isalnum() and not c.isalnum() for c in password)):
        strength_type = 'veryStrong'
    elif (13 <= password_length <= 16) or \
         (8 <= password_length and any(c.isupper() for c in password) and \
          any(c.islower() for c in password) and \
          any(c.isdigit() for c in password)):
        strength_type = 'strong'
    elif (9 <= password_length <= 12) or \
         (6 <= password_length <= 8 and any(c.isdigit() for c in password) and \
          any(c.isalpha() for c in password)):
        strength_type = 'regular'

    return strength_type
